client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -200,7 +212,116 @@ public static void cancelShuffle(int shuffleId, String reason) {
         scheduler.cancelStage(shuffleMapStage.get().id(), new Some<>(reason));
       }
     } else {
-      logger.error("Can not get active SparkContext, skip cancelShuffle.");
+      LOG.error("Can not get active SparkContext, skip cancelShuffle.");
+    }
+  }
+
+  private static final DynFields.UnboundField<ConcurrentHashMap<Long, TaskSetManager>>
+      TASK_ID_TO_TASK_SET_MANAGER_FIELD =
+          DynFields.builder()
+              .hiddenImpl(TaskSchedulerImpl.class, "taskIdToTaskSetManager")
+              .defaultAlwaysNull()
+              .build();
+  private static final DynFields.UnboundField<scala.collection.mutable.HashMap<Long, TaskInfo>>
+      TASK_INFOS_FIELD =
+          DynFields.builder()
+              .hiddenImpl(TaskSetManager.class, "taskInfos")
+              .defaultAlwaysNull()
+              .build();
+
+  protected static TaskSetManager getTaskSetManager(long taskId) {
```

Comment:
We should be very careful with use of `TaskSetManager` - it is not designed to be used outside of the spark scheduler. I have added a specific comment below to mitigate the immediate issue - but within Celeborn we should be careful if we are using this method in future.
Please add a note that there are concerns with use of TSM which need to be analyzed carefully if code evolves and other usages come up.

Reply:
added `VisibleForTesting` and comments